pyews/core.py

```diff
@@ -44,15 +44,32 @@ def invoke(self, soap_body):
                 auth=(self.userconfiguration.credentials.email_address, self.userconfiguration.credentials.password),
                 verify=True
             )
+
+            __LOGGER__.debug('Response HTTP status code: %s', response.status_code)
+            __LOGGER__.debug('Response text: %s', response.text)
+
             parsed_response = BeautifulSoup(response.content, 'xml')
-            try:
-                if parsed_response.find('ResponseCode').string == 'ErrorAccessDenied':
-                    __LOGGER__.info('{}'.format(parsed_response.find('MessageText').string))
-                if parsed_response.find('ResponseCode').string == 'NoError':
-                    return parsed_response
-            except:
-                if parsed_response.find('ErrorCode').string == 'NoError':
-                    return parsed_response
+            if not parsed_response.contents:
+                __LOGGER__.warning(
+                    'The server responded with empty content to POST-request '
+                    'from {current}'.format(current=self.__class__.__name__))
+                return
+
+            response_code = getattr(parsed_response.find('ResponseCode'), 'string', None)
```

Comment:
Fetch response code and error code and then check if anyone of them is "NoError"? So everything went well, return `parsed_response`. If anyone of them is "ErrorAccessDenied"? Log according msg and eventually return none. They are anything else? Log according msg and return none.